### PR TITLE
Terraform 0.14 upgrade

### PR DESCRIPTION
**Pull request by sfc-gh-pkommini**
## what
- Upgrade to support Terraform 0.14 and bring up to current Cloud Posse standard

## why
- Support Terraform 0.14

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -128,7 +128,7 @@ resource "aws_instance" "default" {
 
 module "dns" {
   source  = "cloudposse/route53-cluster-hostname/aws"
-  version = "0.7.0"
+  version = "0.9.0"
```
Comment (maximmi) [inline main.tf:131]: @sfc-gh-pkommini `route53-cluster-hostname` supports context, so please, add
`context = module.this.context` at the end of arguments for module call

**Review by maximmi — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (sfc-gh-pkommini) [reply main.tf:131]: ACK.

Comment (sfc-gh-pkommini) [reply main.tf:131]: @maximmi Seems like these new changes are causing some errors. Maybe I should revert them?

Comment (jamengual) [reply main.tf:131]: no, those are ok, the errors are unrelated

**Comment by maximmi** [comment]
@sfc-gh-pkommini please, undo revert. test fail is unrelated to `context` argument. we will fix the tests on our side

File: main.tf (hunk 1 above)
Comment (sfc-gh-pkommini) [reply main.tf:131]: I see. Will push the changes again.

**Comment by sfc-gh-pkommini** [comment]
Fixed.

**Review by maximmi — APPROVED** [review]

**Pull request merged**